### ISSUE TITLE
PlistQueue incorrectly stores the `IsEmpty` attribute

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/datastructures#52]
The following problem is only visible with PR #49, simply because in the master branch, it is overshadowed by another bug: there we claim that `IsPlistQueueRep` is a collection, but do not implement a `Size` method, just `Length`; but the latter is for lists.
Correcting that, we observe this problematic behavior:

```
gap> q := PlistQueue();
<queue with 0/64 entries>
gap> IsEmpty(q);
true
gap> KnownPropertiesOfObject(q);
[ "IsEmpty" ]
gap> PushFront(q,1);
gap> IsEmpty(q);
true
```

So for some reason, the object remembers that the property `IsEmpty` was true at some point. In contrast, no such problems with e.g. `BinaryHeap` -- and I don't really understand why it works there:
```
gap> h:=BinaryHeap();
<binary heap with 0 entries>
gap> IsEmpty(h);
true
gap> KnownPropertiesOfObject(h);
[  ]
gap> Push(h,1);
gap> IsEmpty(h);
false
```

**Comment by stevelinton** [comment]
The problem is that q does not know that it is mutable. Adding IsMutable to the filters when you construct a PlistQueue solves the problem.

**Comment by fingolfin** [comment]
@stevelinton argh, of course! Fixed in #57

**Comment by markuspf** [comment]
Fixed in #59 (which superseded #57).
